edge-modules/MetricsCollector/IoTHubMetricsSync.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+namespace MetricsCollector
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+
+    public class IoTHubMetricsSync : IMetricsSync
+    {
+        readonly MessageFormatter messageFormatter;
+        readonly Scraper scraper;
+        readonly ModuleClient moduleClient;
+
+        public IoTHubMetricsSync(MessageFormatter messageFormatter, Scraper scraper, ModuleClient moduleClient)
+        {
+            this.messageFormatter = messageFormatter ?? throw new ArgumentNullException(nameof(messageFormatter));
+            this.scraper = scraper ?? throw new ArgumentNullException(nameof(scraper));
+            this.moduleClient = moduleClient;
```

Comment:
check moduleClient for not null

Reply:
I'll also switch to Preconditions for consistency.